observe/datadog.go

```diff
@@ -0,0 +1,96 @@
+package observe
+
+import (
+	datadog "github.com/DataDog/opencensus-go-exporter-datadog"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/pkg/errors"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/trace"
+)
+
+// DatadogExporterConfig provides configuration for the Datadog exporter.
+// Config can be initiated with envconfig from environment
+type DatadogExporterConfig struct {
+	DatadogExporterEnabled        bool   `default:"false" split_words:"true"`
+	DatadogExporterMetricsAddress string `split_words:"true"`
+	DatadogExporterTracesAddress  string `split_words:"true"`
+	DatadogExporterNamespace      string `default:"opencensus" split_words:"true"`
+}
+
+// RegisterAndObserveDatadog will initiate and register Datadog metrics
+// and tracing exporters
+func RegisterAndObserveDatadog(onError func(error)) error {
+	if SkipObserve() {
+		return nil
+	}
+
+	var config DatadogExporterConfig
+	envconfig.Process("", &config)
+
+	exp, err := NewDatadogExporter(config, onError)
+	if err != nil {
+		return errors.Wrap(err, "unable to initiate Datadog's opencensus exporter")
+	}
+	trace.RegisterExporter(exp)
+	view.RegisterExporter(exp)
+
+	return nil
+}
+
+// NewDatadogExporter will return Datadog's opencensus exporter.
+// Exporter will send metrics and traces to Datadog's agent using
+// addresses specified through DatadogExporterConfig
+func NewDatadogExporter(config DatadogExporterConfig, onErr func(error)) (*datadog.Exporter, error) {
+
+	if !config.DatadogExporterEnabled {
+		return nil, errors.New("Datadog exporter disabled")
+	}
+
+	if config.DatadogExporterMetricsAddress == "" && config.DatadogExporterTracesAddress == "" {
+		return nil, errors.New("Missing Datadog agent's adress for metrics and traces")
+	}
+
+	_, service, version := GetServiceInfo()
+
+	opts := getDatadogOpts(config, service, version, onErr)
+
+	return datadog.NewExporter(*opts)
+}
+
+// getDatadogOpts returns Datadog Options that you can pass directly
+// to the OpenCensus exporter or other libraries.
+func getDatadogOpts(config DatadogExporterConfig, service, version string, onErr func(err error)) *datadog.Options {
```

Comment:
simply return `datadog.Options` - no need for the reference (you're using the struct as is later)